src/content/roadmap/account-abstraction/index.md

```diff
@@ -0,0 +1,95 @@
+
+Gas management is also much improved with account abstraction. Not only can applications offer to pay their users' gas fees, but gas fees can be paid in tokens other than ETH, freeing users from having to maintain an ETH balance for funding transactions. This would work by swapping the user's tokens for ETH inside the contract and then using the ETH to pay for gas.
+
+Trusted sessions are also potentially transformative for user experiences, especially for applications like gaming where large numbers of small transactions might need to be approved in a small amount of time. Individually approving each transaction would break the gaming experience, but permanent approval is unsafe. Therefore, the wallet could approve certain transactions for a fixed time, possibly also up to a certain value or only to certain addresses.
+
+It is also interesting to consider how purchases could change with account abstraction. Today, each transaction has to be approved and executed from a wallet pre-funded with a sufficient amount of the correct token. With account abstraction the experience could be more like familiar online shopping where a user could fill a "basket" with items and click once to purchase all at once, with all the logic required handled by the contract, not the user.
+
+These are just a few examples of how user experiences could be levelled up by account abstraction, but there will be many more that we haven't imagined yet. Account abstraction frees developers from the constraints of present-day EOAs, allows them to bring the good aspects of web2 into web3 without sacrificing self-custody and to hack creatively on inventive new user experiences.
+
+## How will account abstraction be implemented? {#how-will-aa-be-implemented}
+
+Smart contract wallets are already implemented today, but they are difficult to implement because they are not supported at the protocol level and instead rely on wrapping relatively complex code around normal Ethereum transactions. This problem can be solved by allowing smart contracts to initiate transactions, since then the logic implemented by the wallet can be handled in Ethereum smart contracts rather than happening off-chain and being translated into Ethereum transactions. This has the added benefit of increasing decentralization since it removes the need for "relayers" run by wallet developers to translate messages signed by the user to normal Ethereum transactions.
```

Comment:
```suggestion
Smart contract wallets exist today but are challenging to implement because the EVM does not support them. Instead, they rely on wrapping relatively complex code around standard Ethereum transactions. Ethereum can change this by allowing smart contracts to initiate transactions, handling the necessary logic in Ethereum smart contracts instead of off-chain. Putting logic into smart contracts also increases Ethereum's decentralization since it removes the need for "relayers" run by wallet developers to translate messages signed by the user to regular Ethereum transactions.
```